dev/tools/localization/gen_l10n.dart

```diff
@@ -512,94 +286,53 @@ String genPluralMethod(Map<String, dynamic> arbBundle, String resourceId) {
         }
       }
-      methodArgs.add("${pluralIds[pluralKey]}: '$argValue'");
+      intlMethodArgs.add("${pluralIds[pluralKey]}: '$argValue'");
     }
   }
 
-  return pluralMethodTemplate
-    .replaceAll('@methodName', resourceId)
-    .replaceAll('@methodParameters', genPluralMethodParameters(placeholders, countPlaceholder, resourceId).join(', '))
-    .replaceAll('@dateFormatting', generateDateFormattingLogic(arbBundle, resourceId))
-    .replaceAll('@numberFormatting', generateNumberFormattingLogic(arbBundle, resourceId))
-    .replaceAll('@intlMethodArgs', methodArgs.join(',\n      '));
-}
-
-String genSupportedLocaleProperty(Set<LocaleInfo> supportedLocales) {
-  const String prefix = 'static const List<Locale> supportedLocales = <Locale>[\n    Locale(';
-  const String suffix = '),\n  ];';
-
-  String resultingProperty = prefix;
-  for (final LocaleInfo locale in supportedLocales) {
-    final String languageCode = locale.languageCode;
-    final String countryCode = locale.countryCode;
-
-    resultingProperty += '\'$languageCode\'';
-    if (countryCode != null)
-      resultingProperty += ', \'$countryCode\'';
-    resultingProperty += '),\n    Locale(';
+  List<String> genPluralMethodParameters([String type]) {
+    return message.placeholders.map((Placeholder placeholder) {
+      final String placeholderType = placeholder.name == countPlaceholder ? 'int' : (type ?? placeholder.type);
+      return '$placeholderType ${placeholder.name}';
+    }).toList();
   }
-  resultingProperty = resultingProperty.substring(0, resultingProperty.length - '),\n    Locale('.length);
-  resultingProperty += suffix;
-
-  return resultingProperty;
-}
-
-bool _isValidClassName(String className) {
-  // Dart class name cannot contain non-alphanumeric symbols
-  if (className.contains(RegExp(r'[^a-zA-Z\d]')))
-    return false;
-  // Dart class name must start with upper case character
-  if (className[0].contains(RegExp(r'[a-z]')))
-    return false;
-  // Dart class name cannot start with a number
-  if (className[0].contains(RegExp(r'\d')))
-    return false;
-  return true;
-}
 
-bool _isNotReadable(FileStat fileStat) {
-  final String rawStatString = fileStat.modeString();
-  // Removes potential prepended permission bits, such as '(suid)' and '(guid)'.
-  final String statString = rawStatString.substring(rawStatString.length - 9);
-  return !(statString[0] == 'r' || statString[3] == 'r' || statString[6] == 'r');
-}
-bool _isNotWritable(FileStat fileStat) {
-  final String rawStatString = fileStat.modeString();
-  // Removes potential prepended permission bits, such as '(suid)' and '(guid)'.
-  final String statString = rawStatString.substring(rawStatString.length - 9);
-  return !(statString[1] == 'w' || statString[4] == 'w' || statString[7] == 'w');
-}
+  if (message.placeholdersRequireFormatting) {
+    return pluralFormatMethodTemplate
+      .replaceAll('@methodName', message.resourceId)
```

Comment:
It seems like you'd want to make sure that there aren't two replacement strings that have the same prefix: For instance "@method" and "@methodParameters".  I know they don't now, but in the future it would be an easy mistake to make.

Maybe have the replacement have a termination character too?  e.g. "@method@" or "{@method}" to make that impossible.  Alternatively, you could replace using a RegExp that adds a `\b` at the end.

Reply:
I used @(foo)